src/timepicker/tests/functional/TimePicker.ts

```diff
@@ -0,0 +1,217 @@
+			.findByCssSelector(`.${comboBoxCss.controls} .${textinputCss.input}`)
+				.type('1')
+			.end()
+			.sleep(DELAY)
+			.execute(`return document.activeElement === document.querySelector('#${exampleId} .${comboBoxCss.controls} .${textinputCss.input}');`)
+			.then(isEqual => {
+				assert.isTrue(isEqual);
+			})
+			.findByCssSelector(`.${comboBoxCss.dropdown}`)
+				.getSize()
+				.then(({ height }) => {
+					assert.isAbove(height, 0);
+				})
+			.end();
+	},
+	'picker opens on focus'() {
+		const { browserName = '' } = this.remote.session.capabilities;
+		if (browserName.toLowerCase() === 'microsoftedge') {
+			this.skip('Edge driver does not handle focus on click');
+		}
+
+		const exampleId = 'example-open-on-focus';
+		return getPage(this.remote, exampleId)
+			.findByCssSelector(`.${comboBoxCss.controls} .${textinputCss.input}`)
+				.click()
+			.end()
+			.sleep(DELAY)
+			.execute(`return document.activeElement === document.querySelector('#${exampleId} .${comboBoxCss.controls} .${textinputCss.input}');`)
+			.then(isEqual => {
+				assert.isTrue(isEqual);
+			})
+			.findByCssSelector(`.${comboBoxCss.dropdown}`)
+				.getSize()
+				.then(({ height }) => {
+					assert.isAbove(height, 0);
+				})
+			.end();
+	},
+	'disabled menu items cannot be clicked'() {
+		const exampleId = 'example-disabled-items';
+		return getPage(this.remote, exampleId)
+			.findByCssSelector(`.${comboBoxCss.controls} .${comboBoxCss.trigger}`)
+				.click()
+			.end()
+			.sleep(DELAY)
+			.execute(`return document.activeElement === document.querySelector('#${exampleId} .${comboBoxCss.controls} .${textinputCss.input}');`)
+			.then(isEqual => {
+				assert.isTrue(isEqual);
+			})
+			.findByCssSelector(`.${comboBoxCss.dropdown}`)
+				.getSize()
+				.then(({ height }) => {
+					assert.isAbove(height, 0);
+				})
+			.end()
+			.findByCssSelector(`.${comboBoxCss.dropdown} .${listboxCss.disabledOption}`)
+				.click()
+			.end()
+			.findByCssSelector(`.${comboBoxCss.controls} .${textinputCss.input}`)
+				.getProperty('value')
+				.then((value) => {
+					assert.strictEqual(value, '', 'The input value should not contain the disabled value.');
+				})
+			.end()
+			.findByCssSelector(`.${comboBoxCss.dropdown}`)
+				.getSize()
+				.then(({ height }) => {
+					assert.isAbove(height, 0, 'The dropdown should remain open.');
+				})
+			.end();
+	},
+	'disabled timepickers cannot be opened'() {
+		const { browserName } = this.remote.session.capabilities;
+		if (browserName === 'firefox') {
+			this.skip('Firefox does not like clicking on disabled things.');
+		}
+
+		return testDisabledPicker(this.remote, 'example-disabled');
+	},
+	'readonly timepickers cannot be opened'() {
+		return testDisabledPicker(this.remote, 'example-readonly');
+	},
+	'validated inputs update on input'() {
+		const { browserName } = this.remote.session.capabilities;
+		if (browserName === 'internet explorer') {
+			this.skip('Test does not work on Internet Explorer');
```

Comment:
Should we be more descriptive with the messages we pass when skipping tests? Do we even know the level at which this test fails? Intern? Leadfoot? A driver? This is largely rhetorical, just trying to think of ways to make the skips less ugly.

This may be as good as we can do for now.

Reply:
When I'm able to easily find a reason in one of the other skips, I use it. Otherwise, trying to figure out the underlying cause is what had been holding up the tests, so I haven't been bothering anymore. I think the plan was to just skip with abandon then document and fix later.